RPLCD/lcd.py

```diff
@@ -94,24 +94,25 @@ def __init__(self, cols=20, rows=4, dotsize=8, charmap='A02', auto_linebreaks=Tr
         self._init_connection()
 
         # Choose 4 or 8 bit mode
-        if self.data_bus_mode == c.LCD_4BITMODE:
-            # Hitachi manual page 46
-            self.command(0x03)
-            c.msleep(4.5)
-            self.command(0x03)
-            c.msleep(4.5)
-            self.command(0x03)
-            c.usleep(100)
-            self.command(0x02)
-        elif self.data_bus_mode == c.LCD_8BITMODE:
-            # Hitachi manual page 45
-            self.command(0x30)
-            c.msleep(4.5)
-            self.command(0x30)
-            c.usleep(100)
-            self.command(0x30)
-        else:
-            raise ValueError('Invalid data bus mode: {}'.format(self.data_bus_mode))
+        if self._i2c_expander != 'JHD1804':
```

Comment:
Why is this not applicable to the JHD1804? The way I understand it, a port expander only forwards the commands to the hitachi chip.

Reply:
Also, this line doesn't work, see tests

Reply:
this is not an ordinary i2c expander, but a full driver with built-in i2c, a replacement for the hitachi chip, usually compatible, but the initialization looks different because you do not need to set the 4 or 8 bit communication function:
![obraz](https://github.com/user-attachments/assets/ac7b789f-9046-4687-9d89-6e14c6c865fa)

Reply:
this line doesn't work in tests, but it works in reality. the value of `_i2c_expander` comes from the `CharLCD` class, the same as e.g. in the next line `if self.data_bus_mode == c.LCD_4BITMODE:`, with the difference that `data_bus_mode` is always defined `self.data_bus_mode = c.LCD_4BITMODE` _(line 153 in i2c.py)_ while the expander must be defined when creating the object:
```
if i2c_expander in ['PCF8574', 'MCP23008', 'MCP23017', 'JHD1802']:
 self._i2c_expander = i2c_expander
 else:
 raise NotImplementedError('I2C expander "%s" is not supported.' % i2c_expander)
```
_(i2c.py 131 to 134)_
I don't see an expander defined anywhere in the tests, but the object creation still goes through?